src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -178,6 +199,21 @@ export default function getAdaptationSwitchStrategy(
                                { type: "continue", value: undefined };
 }
 
+/**
+ * Returns `true` if at least one codec of the Representations in the given
+ * Adaptation has a codec compatible with the given SegmentBuffer.
+ * @param {Object} adaptation
+ * @param {Object} segmentBuffer
+ * @returns {boolean}
+ */
+function hasCompatibleCodec(
+  adaptation : Adaptation,
+  segmentBuffer : SegmentBuffer<unknown>
+) : boolean {
+  return adaptation.getPlayableRepresentations().some(rep =>
+    areCodecsCompatible(rep.getMimeTypeString(), segmentBuffer.codec ?? ""));
```

Comment:
Maybe we could just return true if the `segmentBuffer.codec` property is equal to `""`.

Reply:
I think you're right.

I'm even wondering if we're not under the risk of reloading when switching to a texttrack in another format under the current implementation - which would be unnecessary.

But a `segmentBuffer.codec` set to `undefined` is for now described as:
```
   * `undefined` either if unknown or if the codec does not matter for this
   * SegmentBuffer implementation.
```
For the "not matter" part returning `true` would be OK but for the "unknown" part less so.

I have to look at the real reasons for SegmentBuffer.codec being `undefined` to see if the "unknown" part could be removed from the documentation.

Reply:
Maybe that checking first that we're either playing audio or video is better for the text track case.

But you're still right that something has to be done for the codec being `undefined` here

Reply:
Done